ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -16,11 +18,88 @@ use std::{
 use typed_index_collections::TiVec;
 use ykaddr::addr::symbol_vaddr;
 
-// Since the AOT versions of these data structures contain no AOT/JIT-IR-specific indices we can
-// share them. Note though, that their corresponding index types are not shared.
-pub(crate) use super::aot_ir::IntegerType;
+// This is simple and can be shared across both IRs.
 pub(crate) use super::aot_ir::Predicate;
 
+/// A fixed-width integer type.
+///
+/// Signedness is not specified. Interpretation of the bit pattern is delegated to operations upon
+/// the integer.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(crate) struct IntegerType {
+    num_bits: u32,
```

Comment:
Do we really need `u32` bits?!

Reply:
Probably not, but that's how it is everywhere else in our IRs. We semi-inherited this from llvm. They use `unsigned`, which on x86_64 is u32...

We can change it, but I suggest not in this PR. Maybe a task for the wolf?

Reply:
Fair enough, this is something we can rejig later.